scipy/fftpack/basic.py

```diff
@@ -147,14 +147,17 @@ def _asfarray(x):
 def _fix_shape(x, n, axis):
     """ Internal auxiliary function for _raw_fft, _raw_fftnd."""
     s = list(x.shape)
+    if axis >= len(s) or axis < - len(s):
+        raise IndexError("Input array does not have specified axis.")
+    axis = axis % len(s)
     if s[axis] > n:
-        index = [slice(None)]*len(s)
-        index[axis] = slice(0,n)
+        index = tuple(slice(None) if i != axis else slice(0, n)
+                      for i in range(len(s)))
         x = x[index]
```

Comment:
I'd expect it to be faster to just do `x[tuple(index)]` here - numpy < 1.13 did that internally anyway